### ISSUE TITLE
Move controller specs into requests spec folder

### ISSUE THREAD

**Issue by sethherr** [bikeindex/bike_index#845]
As per https://github.com/bikeindex/bike_index/pull/837#discussion_r290419477, we're moving toward request specs.

Rather than having multiple places to look for things, move all the controller specs into request specs and gradually transition them over.

**Comment by sethherr** [comment]
The one issue with consolidating into request specs (rather than moving request specs into controller specs) is that when moving the controller specs, we will need to specify that they are controller specs - which will be a larger number of additions

**Comment by sethherr** [comment]
From rspec-rails response to ["Are controller specs discouraged in favor of request specs?"](https://github.com/rspec/rspec-rails/issues/1838)

> The best advice we can give for people working on rails apps is to write request specs instead of controller specs. I'd consider writing a new controller spec in 2017 to be a smell, even on an older rails app. Keeping controller specs around for legacy reasons is absolutely fine though.

I'm updating and renaming this issue to reflect this. I don't think we should move the controller specs into the request specs folder, because they are different and removing all of the tests that use `assign` as an important part of their test functionality would be a loss.

I'm going to call controller specs deprecated, no new controller specs should be added and whenever touching existing controller specs, strongly consider moving them into request specs. Let me know if that sounds good @jmromer ?

**Comment by jmromer** [comment]
I don't think I follow, but it's less time-consuming to let the code speak than disambiguate natural language in this case — have a look at https://github.com/bikeindex/bike_index/pull/848, and feel free to close if it doesn't seem like a good move right now. Definitely won't mind :)

**Comment by sethherr** [comment]
The issue I see with just moving the specs around is that controller specs should no longer be used (according to ruby and rspec, visible in the ["are controller specs discouraged"](https://github.com/rspec/rspec-rails/issues/1838)) - so renaming our request specs into controller specs makes us look like we aren't following current best practices. I think it will make it easier to find specs at the cost of making it explicit about which thing is which (and how far we are along on this migration).

Immediately refactoring the controller specs into request specs seems pretty low priority to me. I would prefer the incremental approach, because it means that, hopefully, we'll be focused on the areas that we're refactoring the tests for when we refactor them (because we'll be touching the area).

The disadvantage I see to not immediately refactoring means that this "No new controller specs should be added and whenever touching existing controller specs, strongly consider moving them into request specs" is institutional knowledge rather than code :/

**Comment by jmromer** [comment]
> so renaming our request specs into controller specs makes us look like we aren't following current best practices.

My position is that the location or the name of the file doesn't / shouldn't determine whether it's a request or controller spec, the `type:` annotation and how the test is written does (it would not be the first time I've seen "controller" specs written as request specs and housed in `spec/controllers`), but I agree I don't think this is worth dwelling on much.

**Comment by sethherr** [comment]
This is easily possible when we upgrade Rails, since you can check `assigns` in rails 5 request specs. So we're waiting on that before worrying about this.
